### ISSUE TITLE
Unexpected errors with `willReturn` called on `with`

### ISSUE THREAD

**Issue by alexander-schranz** [phpstan/phpstan-phpunit#299]
After upgrading to PHPStan 2.2.0 the `willReturn` on the `with` seems not longer be detected correctly:

```php
use PHPUnit\Framework\MockObject\MockObject;
use PHPUnit\Framework\TestCase;
use Symfony\Component\DependencyInjection\ContainerBuilder;

class RegisterPreviewWebspaceClassPassTest extends TestCase
{
    /** @var ContainerBuilder&MockObject */
    private $containerBuilderMock;

    public function testShouldNotOverrideWebspaceCacheClass(): void
    {
        $this->containerBuilderMock
            ->method('hasExtension')
            ->with('sulu_core')
            ->willReturn(false);
    }
```

```
  28     Call to an undefined method PHPUnit\Framework\MockObject\InvocationStubber::with().
         🪪  method.notFound
```

Packages which are installed related to PHPUnit:

```
phpspec/prophecy-phpunit                         2.5.0   Integrating the Prophecy mocking library in PHPUnit test cases
phpstan/phpstan-phpunit                          2.0.16  PHPUnit extensions and rules for PHPStan
phpunit/php-code-coverage                        14.1.9  Library that provides collection, processing, and rendering functionality for PHP code covera...
phpunit/php-file-iterator                        7.0.0   FilterIterator implementation that filters files based on a list of suffixes.
phpunit/php-invoker                              7.0.0   Invoke callables with a timeout
phpunit/php-text-template                        6.0.0   Simple template engine.
phpunit/php-timer                                9.0.0   Utility class for timing
phpunit/phpunit                                  13.1.13 The PHP Unit Testing framework.
symfony/phpunit-bridge                           7.4.8   Provides utilities for PHPUnit, especially user deprecation notices management
```

**Comment by VincentLanglet** [comment]
> After upgrading to PHPStan 2.2.0 the willReturn on the with seems not longer be detected correctly:

Are you sure that only PHPStan was bump ? Not the extension ? Not Phpunit ?

Also it make me think of
- https://github.com/sebastianbergmann/phpunit/issues/6504
- https://github.com/sebastianbergmann/phpunit/issues/6507
- https://github.com/sebastianbergmann/phpunit/issues/6508

So using `with` on a mock without `expect` is/will be wrong

**Comment by alexander-schranz** [comment]
@VincentLanglet you are right seems not the PHPStan 2.2.0 release, unsure which package it is.
